staging_vespalib/src/vespa/vespalib/metrics/dummy_metrics_manager.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include <memory>
+#include <thread>
+#include <vespa/vespalib/stllike/string.h>
+#include "name_collection.h"
+#include "current_samples.h"
+#include "snapshots.h"
+#include "metrics_manager.h"
+#include "clock.h"
+
+namespace vespalib {
+namespace metrics {
+
+/**
+ * Dummy manager that discards everything, use
+ * for unit tests where you don't care about
+ * metrics.
+ **/
+class DummyMetricsManager : public MetricsManager
+{
+private:
+    DummyMetricsManager() {}
+public:
+    ~DummyMetricsManager();
+
+    static std::shared_ptr<MetricsManager> create() {
+        return std::shared_ptr<MetricsManager>(new DummyMetricsManager());
```

Comment:
cannot do that since make_shared can't use the private constructor.  Also I really want to make a shared_ptr of MetricsManager but use the DummyMetricsManager constructor, so I think this code has better clarity.